packages/router/__tests__/router-memory-test.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @jest-environment node
+ */
```

Comment:
Because tests by default run in jsdom, we didn't catch this regression since `window` was defined during test runs when using a memory history.  Now we have this standalone file that runs directly in node to catch those